### PR TITLE
Adding azure vnet peering test.

### PR DESCRIPTION
**Pull request by kwoodson**
Adding vnet testing to the production RP.

1. Test creates a resource group with a vnet.
2. Creates a cluster with the vnetpeerid set in the networkprofile
3. Verifies the vnets are peered.

### CODE REVIEW

File: ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-periodics.yaml [hunk 1]
```diff
@@ -416,6 +416,61 @@ periodics:
...
+  interval: 24h
+  name: periodic-ci-azure-vnet-e2e
+  skip_cloning: true
+  spec:
+    containers:
+    - args:
+      - --git-ref=openshift/openshift-azure@master
+      - --secret-dir=/usr/local/e2e-secrets/azure
+      - --target=[output:stable:azure-controllers]
+      - --target=[output:stable:etcdbackup]
```
[the remark this answers is not in the document]
Comment (charlesakalugwu) [reply ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-periodics.yaml]: yeah @kwoodson you need to include include the [output] targets if those images are needed by your test

[the remark this answers is not in the document]
Comment (kwoodson) [reply ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-periodics.yaml]: The only thing I'm currently testing is that the cluster create is successful and that the vnet created is peered with the incoming peerVnetID passed in.

Should I remove these then?

[the remark this answers is not in the document]
Comment (kargakis) [reply ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-periodics.yaml]: All of these images are used by the fake RP to deploy the cluster. You are using the production RP, essentially these images are not actually used by your test.

[the remark this answers is not in the document]
Comment (kwoodson) [reply ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-periodics.yaml]: Removed.

**Comment by kwoodson** [comment]
This relies on https://github.com/openshift/openshift-azure/pull/886

**Pull request merged**